nautobot/extras/jobs.py

```diff
@@ -979,23 +989,37 @@ def run_job(data, request, job_result_pk, commit=True, *args, **kwargs):
         job_result.completed = timezone.now()
         job_result.save()
         return False
-    job = job_class()
-    job.job_result = job_result
 
-    # Capture the file IDs for any FileProxy objects created so we can cleanup later.
-    file_fields = list(job._get_file_vars())
-    file_ids = [data[f] for f in file_fields]
-
-    # Attempt to resolve serialized data back into original form by creating querysets or model instances
-    # If we fail to find any objects, we consider this a job execution error, and fail.
-    # This might happen when a job sits on the queue for a while (i.e. scheduled) and data has changed
-    # or it might be bad input from an API request, or manual execution.
+    try:
+        job = job_class()
+        job.active_test = "initialization"
+        job.job_result = job_result
+    except Exception as e:
+        logger.error("Error initializing job object.")
+        logger.error(e)
+        stacktrace = traceback.format_exc()
+        job_result.log_failure(f"Error initializing job:\n```\n{stacktrace}\n```")
```

Comment:
Whoops, I think I may have led you astray with my previous comments. `log_failure()` is an API on `Job`, not on `JobResult`. Since you don't have a `Job` here, you need to call `job_result.log()` instead:

```suggestion
        job_result.log(
            f"Error initializing job:\n```\n{stacktrace}\n```",
            level_choice=LogLevelChoices.LOG_FAILURE,
            grouping="initialization",
            logger=logger,
        )
```